include/cudaq/Support/Device.h

```diff
@@ -0,0 +1,173 @@
+/****************************************************************-*- C++ -*-****
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#pragma once
+
+#include "cudaq/ADT/GraphCSR.h"
+#include "cudaq/Support/Graph.h"
+
+namespace cudaq {
+
+class Device {
```

Comment:
This class is probably worthy of some documentation.

Reply:
Updated in 9f60585e